scripts/lib/CIME/build.py

```diff
@@ -390,7 +391,13 @@ def _build_libraries(case, exeroot, sharedpath, caseroot, cimeroot, libroot, lid
 
     # Build shared code of CDEPS nuopc data models
     build_script = {}
-    if comp_interface == "nuopc" and (not ufs_driver or ufs_driver != 'nems'):
+    if (comp_interface == "nuopc" and (not ufs_driver or ufs_driver != 'nems')
+        and lilac_mode != 'on'):
```

Comment:
Note that LILAC_MODE is undefined when not using CTSM, but that's okay: lilac_mode will end up as None, which gives equivalent results to 'off' (i.e., not 'on').